### PR TITLE
add sensu version facter

### PR DESCRIPTION
**Pull request by bovy89**

### CODE REVIEW

File: lib/facter/sensu_version.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Facter.add(:sensu_version) do
+  confine :kernel => 'Linux'
+  setcode do
+    if File.exists? '/opt/sensu/embedded/bin/gem'
+      sensuversion = Facter::Util::Resolution.exec('/opt/sensu/embedded/bin/gem list "^sensu$"')
```
Comment (ghoneycutt) [inline lib/facter/sensu_version.rb]: Suggest we use `/opt/sensu/bin/sensu-client --version`.

**Comment by ghoneycutt** [comment]
Thank you for the contribution!

We need spec tests for this, as well as ensure it does not error when used on a system that does not have sensu installed.

File: spec/unit/facter/sensu_version_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    describe 'returns nil when sensu is not present' do
+        it do
+          File.stubs(:exists?).with('/opt/sensu/embedded/bin/sensu-client').returns(false)
+          expect(Facter.value(:sensu_version)).to be_nil
+        end
+    end
+  end
+
+  context "on non linux systems" do
```
Comment (ghoneycutt) [inline spec/unit/facter/sensu_version_spec.rb]: I believe this should read non-posix systems as the test should work on Solaris, OSX and FreeBSD.

File: lib/facter/sensu_version.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Facter.add(:sensu_version) do
+  confine :kernel => 'Linux'
```
Comment (ghoneycutt) [inline lib/facter/sensu_version.rb]: This should not confine for Linux as it should work on the other posix platforms.

Perhaps a conditional such as
```
if Facter.value('kernel') == 'windows'
  <GH will figure out windows code>
else
  <posix code here>
end
```

**Comment by ghoneycutt** [comment]
Thanks for adding the tests! Unless you're keen to add windows support, I'll handle that and then merge. Want to ensure that all of our platforms can use this great new fact.

**Comment by ghoneycutt** [comment]
We should add `facter -p sensuversion` to the vagrant scripts. This will test that this works while also giving us useful information.

**Comment by treydock** [comment]
@ghoneycutt Changes made in #894

**Comment by ghoneycutt** [comment]
Released functionality in v2.52.0

**Pull request closed without merging**